--- a/net/minecraft/client/MainWindow.java
+++ b/net/minecraft/client/MainWindow.java
@@ -101,7 +101,7 @@
             GLFW.glfwWindowHint(135181, Config.getAntialiasingLevel());
         }
 
-        this.handle = GLFW.glfwCreateWindow(this.width, this.height, "Minecraft 1.13.2", this.fullscreen ? this.monitor.getMonitorPointer() : 0L, 0L);
+        this.handle = GLFW.glfwCreateWindow(this.width, this.height, mcIn.minecriftVerString, this.fullscreen ? this.monitor.getMonitorPointer() : 0L, 0L);
         mcIn.isWindowFocused = true;
         this.setMonitorFromVirtualScreen();
         GLFW.glfwMakeContextCurrent(this.handle);
