--- a/net/minecraft/client/Minecraft.java
+++ b/net/minecraft/client/Minecraft.java
@@ -200,6 +200,7 @@
 import org.lwjgl.glfw.GLFW;
 import org.lwjgl.glfw.GLFWErrorCallback;
 import org.lwjgl.glfw.GLFWErrorCallbackI;
+import org.vivecraft.render.PlayerModelController;
 
 public class Minecraft implements IThreadListener, ISnooperInfo, IGuiEventListenerDeferred
 {
@@ -1453,7 +1454,11 @@
         this.entityRenderer.getMouseOver(1.0F);
         this.tutorial.onMouseHover(this.world, this.objectMouseOver);
         this.profiler.startSection("gameMode");
-
+        
+        //Vivecraft
+	    PlayerModelController.getInstance().tick();
+	    //
+	    
         if (!this.isGamePaused && this.world != null)
         {
             this.playerController.tick();
@@ -2726,4 +2731,16 @@
     {
         return this.fontResourceMananger;
     }
+    
+    //Vivecraft Additions
+    
+    public final String minecriftVerString = "Vivecraft 1.13.2 jrbudda-NONVR-1-r9";
+
+    public void printChatMessage(String msg)
+    {
+    	if (this.world != null) {
+    		ITextComponent chatText = new TextComponentString(msg);
+    		this.ingameGUI.getChatGUI().printChatMessage(chatText);
+    	}
+    }
 }
