--- a/net/minecraft/client/tutorial/Tutorial.java
+++ b/net/minecraft/client/tutorial/Tutorial.java
@@ -97,7 +97,9 @@
             this.stop();
         }
 
-        this.tutorialStep = this.minecraft.gameSettings.tutorialStep.create(this);
+        // VIVECRAFT: Real men don't need tutorials.
+        //this.tutorialStep = this.minecraft.gameSettings.tutorialStep.create(this);
+        this.tutorialStep = TutorialSteps.NONE.create(this);
     }
 
     public void tick()
