--- a/net/minecraft/client/renderer/entity/layers/LayerHeldItem.java
+++ b/net/minecraft/client/renderer/entity/layers/LayerHeldItem.java
@@ -1,13 +1,19 @@
 package net.minecraft.client.renderer.entity.layers;
 
+import org.vivecraft.render.PlayerModelController;
+import org.vivecraft.render.PlayerModelController.RotInfo;
+
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.client.renderer.block.model.ItemCameraTransforms;
 import net.minecraft.client.renderer.entity.RenderLivingBase;
 import net.minecraft.client.renderer.entity.model.ModelBiped;
+import net.minecraft.client.renderer.entity.model.ModelPlayerVR;
 import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.EnumHandSide;
+import net.minecraft.util.math.Vec3d;
 
 public class LayerHeldItem implements LayerRenderer<EntityLivingBase>
 {
@@ -43,23 +49,46 @@
 
     private void renderHeldItem(EntityLivingBase p_188358_1_, ItemStack p_188358_2_, ItemCameraTransforms.TransformType p_188358_3_, EnumHandSide handSide)
     {
-        if (!p_188358_2_.isEmpty())
-        {
-            GlStateManager.pushMatrix();
-            this.translateToHand(handSide);
-
-            if (p_188358_1_.isSneaking())
-            {
-                GlStateManager.translatef(0.0F, 0.2F, 0.0F);
-            }
-
-            GlStateManager.rotatef(-90.0F, 1.0F, 0.0F, 0.0F);
-            GlStateManager.rotatef(180.0F, 0.0F, 1.0F, 0.0F);
-            boolean flag = handSide == EnumHandSide.LEFT;
-            GlStateManager.translatef((float)(flag ? -1 : 1) / 16.0F, 0.125F, -0.625F);
-            Minecraft.getMinecraft().getFirstPersonRenderer().renderItemSide(p_188358_1_, p_188358_2_, p_188358_3_, flag);
-            GlStateManager.popMatrix();
-        }
+    	if (!p_188358_2_.isEmpty())
+    	{
+    		boolean vr =false;;
+    		RotInfo rotinfo = null;
+    		if(p_188358_1_ instanceof EntityPlayer){
+    			if(PlayerModelController.getInstance().isTracked(p_188358_1_.getUniqueID())){
+    				rotinfo = PlayerModelController.getInstance().getRotationsForPlayer(p_188358_1_.getUniqueID());
+    				if(rotinfo.seated ==false){
+    					vr = true;
+    				}
+    			}
+    		}
+
+    		GlStateManager.pushMatrix();
+
+    		if (p_188358_1_.isSneaking())
+    		{
+    			GlStateManager.translatef(0.0F, 0.2F, 0.0F);
+    		}
+
+    		this.translateToHand(handSide);
+
+    		GlStateManager.rotatef(-90.0F, 1.0F, 0.0F, 0.0F);
+    		GlStateManager.rotatef(180.0F, 0.0F, 1.0F, 0.0F);
+    		boolean flag = handSide == EnumHandSide.LEFT;
+
+    		//VIVECRAFT
+    		if(!vr) {
+    			GlStateManager.translatef((float)(flag ? -1 : 1) / 16.0F, 0.125F, -0.625F);   
+    		}
+    		else{
+    			GlStateManager.translatef((float)(flag ? -1 : 1) / 16.0F, 0.125F,  -.2125f);
+    			GlStateManager.rotatef(-80.0F, 1.0F, 0.0F, 0.0F);        		
+    			GlStateManager.translatef(0, 0.155F, -0.215F);
+    		}
+    		//
+    		
+    		Minecraft.getMinecraft().getFirstPersonRenderer().renderItemSide(p_188358_1_, p_188358_2_, p_188358_3_, flag);
+    		GlStateManager.popMatrix();
+    	}
     }
 
     protected void translateToHand(EnumHandSide p_191361_1_)
