--- a/net/minecraft/client/renderer/BufferBuilder.java
+++ b/net/minecraft/client/renderer/BufferBuilder.java
@@ -69,7 +69,7 @@
         {
             int i = this.byteBuffer.capacity();
             int j = i + MathHelper.roundUp(increaseAmount, 2097152);
-            LOGGER.debug("Needed to grow BufferBuilder buffer: Old size {} bytes, new size {} bytes.", i, j);
+            LOGGER.info("Needed to grow BufferBuilder buffer: Old size {} bytes, new size {} bytes.", i, j);
             int k = this.rawIntBuffer.position();
             ByteBuffer bytebuffer = GLAllocation.createDirectByteBuffer(j);
             this.byteBuffer.position(0);
